### ISSUE TITLE
Using `pairdist.default` for single vector

### ISSUE THREAD

**Issue by rubak** [spatstat/spatstat#81]
Recently @adchSTATS mentioned to me, that he had tried `pairdist.default` to calculate pairwise distances in 1-D, which didn't work as expected. I agree that the documentation is slightly unclear about the fact that it only works in 2-D. Should we update documentation or change behaviour?
Currently, the following happens:
```
> library(spatstat)
> x <- c(2, 4)
> all.equal(pairdist(x), pairdist(1:2, x))
[1] TRUE
```
I.e. the missing coordinates are set to `1:n`. In my opinion it would be better to calculate the 1-D distances or generate an error. Am I missing a use case?

**Comment by baddstats** [comment]
Oh dear. Yes, we should make it work in 1D as well.

**Comment by baddstats** [comment]
I found it difficult to modify `pairdist.default` to work for 1-dimensional coordinates while at the same time preventing it from doing other unexpected things. It seems easier to refuse to handle data that are not 2-dimensional. I've implemented that in the current version of `spatstat`. If you don't like this, we can easily make it work for 1-D coordinates, but then I would want advice about how to handle k-dimensional coordinates for k > 2.
